### PR TITLE
feat(core): disable trigger

### PR DESCRIPTION
**Pull request by Skraye**

### CODE REVIEW

File: core/src/main/java/io/kestra/core/models/triggers/Trigger.java [hunk 1]
```diff
@@ -201,6 +204,7 @@ public static Trigger of(Flow flow, AbstractTrigger abstractTrigger, ConditionCo
             .triggerId(abstractTrigger.getId())
             .date(ZonedDateTime.now().truncatedTo(ChronoUnit.SECONDS))
             .nextExecutionDate(((PollingTriggerInterface) abstractTrigger).nextEvaluationDate(conditionContext, lastTrigger))
+            .disabled(lastTrigger.stream().map(Trigger::getDisabled).findFirst().orElse(false))
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/models/triggers/Trigger.java:207]:
```suggestion
            .disabled(lastTrigger.map(t -> t.getDisabled()).orElse(Boolean.FALSE))
```

File: core/src/main/java/io/kestra/core/models/triggers/TriggerContext.java [hunk 2]
```diff
@@ -44,6 +43,9 @@ public class TriggerContext {
     @Nullable
     private Backfill backfill;
 
+    @Builder.Default
+    private Boolean disabled = false;
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/models/triggers/TriggerContext.java:47]:
```suggestion
    private Boolean disabled = Boolean.FALSE;
```

**Review by loicmathieu — APPROVED** [review]
LGTM, I made 2 code improvement suggestion but you can merge it as is if you want.

I didn't QA it but as I work on other features on trigger I'll rebase mine and will QA it during my work.

**Comment by Skraye** [comment]
closes #2912

**Pull request merged**